transform/transformer.go

```diff
@@ -146,6 +244,38 @@ func (tr *Transformer) walker(path string, value json.RawMessage) error {
 	return nil
 }
 
+// walker is a WalkFunc for the Transformer which builds an representation of the fields and transforms in the schema.
+// This is later used to do the actual transform for incoming data
+func (tr *TransformerXML) walker(path string, value json.RawMessage) error {
```

Comment:
Also looking nearly identical, maybe the thing to do is to combine the transformers adding a field that says if the input is XML or JSON and then just pass that down to the new*Transformer functions.

Reply:
I like this idea